docs/faq/DesignDecisions.md

```diff
@@ -0,0 +1,75 @@
+The [curried function signature](https://github.com/reactjs/redux/issues/1744) of declaring middleware is [deemed unnecessary](https://github.com/reactjs/redux/pull/784) by some, because both store and next are available when the applyMiddleware function is executed. This issue has been determined to not be [worth introducing breaking changes](https://github.com/reactjs/redux/issues/1744).
+
+<a id="closure-dispatch"></a>
+### Why does applyMiddleware use a closure for dispatch?
+applyMiddleware takes the existing dispatch from the store and closes over it to create the initial chain of middlewares that have been invoked with an object that exposes the getState and dispatch functions, which enables middlewares that [rely on dispatch during initialization](https://github.com/reactjs/redux/pull/1592) to run. 
+
+<a id="combineReducers-limitations"></a>
+### Can you please change combineReducers to support nested state trees?
+No, but there are some limits to combineReducers that are worth knowing.
+- combineReducers receives an object where the values are all reducer names. Additional nesting is not possible, which limits the state shape. The following code is not possible:
+```
+const rootReducer = combineReducers({
+    a : reducerA,
+    b : {
+        b1 : reducerB1,
+        b2 : reducerB2
+    }
+});
+```
+- reducers within combineReducers are only passed the part of the state that they modify, and not the top state.
+
+The default utility combineReducers is only one way to build a complex reducer. Consider using libraries like [combineSectionReducers](https://github.com/ryo33/combine-section-reducers) or [reduceReducers](https://github.com/acdlite/reduce-reducers) if you want your reducers to have a nested tree structure.
+
+<a id="no-asynch-in-mapDispatchToProps"></a>
+### Why doesn't mapDispatchToProps allow use of return values from getState() or mapStateToProps()?
+In general, connect generates a props object out of a closure that is injected with both the current state and dispatch. Asynchronous logic does not belong in the mapStateToProps and mapDispatchToProps functions at all. They should be only pure functions which transform the state to props and bind action creators to dispatch. 
+
+You cannot modify the state during the execution of mapStateToProps. Modifying the state from these functions could lead to infinite loops because every update would reinvoke the map functions. Calling getState inside mapStateToProps would always just return the same state that is passed to the function. 
+
+The preferred way to handle this use-case (needing to alter props based on the current state and mapDispatchToProps functions) is to work from  mergeProps, the third argument to the connect function. If specified, it is passed the result of mapStateToProps(), mapDispatchToProps(), and the container component's props. The plain object returned from mergeProps will be passed as props to the wrapped component.
+
+#### Further information
```

Comment:
I'd like these links moved to each be at the end of the appropriate question/answer.